src/vaults/NftGate.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC721} from "solmate/src/tokens/ERC721.sol";
+import {AffineGovernable} from "src/utils/AffineGovernable.sol";
+
+abstract contract NftGate is AffineGovernable {
+    ERC721 public accessNft;
+    uint16 public withdrawalFeeWithNft;
+    bool needNftToDeposit;
```

Comment:
Can be public.

Reply:
Leaving these as is for now because of the code size issue. Making these public pushes `L2VaultV2` over the limit.